From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Etaash Mathamsetty
 <45927311+Etaash-mathamsetty@users.noreply.github.com>
Date: Sat, 18 Apr 2026 01:55:02 -0500
Subject: [PATCH 09/46] 
 0013-winewayland-Ensure-no-negative-viewport-destinations.patch

---
 dlls/winewayland.drv/wayland_surface.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/dlls/winewayland.drv/wayland_surface.c b/dlls/winewayland.drv/wayland_surface.c
index ce4e0883854..da118bb1b0a 100644
--- a/dlls/winewayland.drv/wayland_surface.c
+++ b/dlls/winewayland.drv/wayland_surface.c
@@ -690,7 +690,7 @@ static void wayland_surface_reconfigure_size(struct wayland_surface *surface,
 {
     TRACE("hwnd=%p size=%dx%d\n", surface->hwnd, width, height);
 
-    if (width != 0 && height != 0)
+    if (width > 0 && height > 0)
         wp_viewport_set_destination(surface->wp_viewport, width, height);
     else
         wp_viewport_set_destination(surface->wp_viewport, -1, -1);
@@ -721,7 +721,7 @@ static void wayland_surface_reconfigure_client(struct wayland_surface *surface,
         wl_subsurface_place_above(client->wl_subsurface, surface->wl_surface);
     }
 
-    if (rect.left != rect.right && rect.top != rect.bottom)
+    if (rect.right > rect.left && rect.bottom > rect.top)
         wp_viewport_set_destination(client->wp_viewport, rect.right - rect.left, rect.bottom - rect.top);
     else /* We can't have a 0x0 destination, use 1x1 instead. */
         wp_viewport_set_destination(client->wp_viewport, 1, 1);
